InteractiveHtmlBom/ecad/fusion_eagle.py

```diff
@@ -0,0 +1,440 @@
+                ratio = 0.08 if 'ratio' not in attr.attrib else float(attr.attrib['ratio'])/100
+                dwg = {'type': 'text',
+                       'text': text,
+                       'pos': [xpos, ypos],
+                       'height': size,
+                       'width': size,
+                       'justify': justify,
+                       'thickness': size * ratio,
+                       'attr': [] if not mirrored else ['mirrored'],
+                       'angle': angle
+                       }
+
+                self.font_parser.parse_font_for_string(text)
+                for (c, wl) in self.font_parser.get_parsed_font().items():
+                    if c not in self.pcbdata['font_data']:
+                        self.pcbdata['font_data'][c] = wl
+                if mirrored:
+                    self.pcbdata['drawings']['silkscreen']['B'].append(dwg)
+                else:
+                    self.pcbdata['drawings']['silkscreen']['F'].append(dwg)
+
+    @staticmethod
+    def _segments_to_polygon(segs):
+        polygon = []
+        for vertex in segs.iter('vertex'):
+            polygon.append([float(vertex.attrib['x']), -float(vertex.attrib['y'])])
+        return polygon
+
+    def _add_zone(self, poly, net):
+        dest = self.pcbdata['zones']['F'] if poly.attrib['layer'] == '1' else self.pcbdata['zones']['B']
+        if poly.tag == 'polygonpour':
+            segs = poly.find('polygonfilldetails').find('polygonshape').find('polygonoutlinesegments')
+        else:
+            segs = poly
+        zone = {'net': net,
+                'polygons': []}
+        zone['polygons'].append(self._segments_to_polygon(segs))
+        dest.append(zone)
+
+    def parse(self):
+        ext = splitext(self.file_name)[1]
+        if ext.lower() == '.fbrd':
+            with zipfile.ZipFile(self.file_name) as myzip:
+                brdfilename = [fname for fname in myzip.namelist() if splitext(fname)[1] == '.brd']
+                with myzip.open(brdfilename[0]) as brdfile:
+                    return self._parse(brdfile)
+        elif ext.lower() == '.brd':
+            with open(self.file_name, 'r', encoding='utf-8') as brdfile:
+                return self._parse(brdfile)
+
+    def _parse(self, brdfile):
+        brdxml = ElementTree.parse(brdfile)
+        root = brdxml.getroot()
+        board = root.find('drawing').find('board')
+        plain = board.find('plain')  # contains stuff not associated with a net (silkscreen, holes)
+        elements = board.find('elements')
+        signals = board.find('signals')
+        mv = [el.attrib['value'] for el in root.iter('param') if el.attrib['name'] == 'rlMinViaOuter'][0]
+        mv_val = float(''.join(d for d in mv if d in digits))
+        mv_units = (''.join(d for d in mv if d in ascii_lowercase))
+        if mv_units == 'mm':
+            self.min_via_w = mv_val * 0.001
+        elif mv_units == 'mil':
+            self.min_via_w = mv_val * 0.0254
```

Comment:
It seems weird to me that you have to care for units only here and nowhere else.
Are all the other dimensions in microns and just this design rule is specified with units?

The constants here confuse me too. If min via width is given in mm then it's 1/1000th of other coordinates so they are given in meters? But if it's given in mils then it's 0.0254 so other coordinates are in millimeters?

Keep in mind that ibom rounds all floating point numbers to 6 decimal digits in one of last steps to reduce the json size and trim floating point errors induced artifacts. If coordinates are in meters then it makes sense to multiply them by 10^3 to have sufficient precision.

Reply:
The Eagle/Fusion BRD file format implicitly uses units of millimeters _almost_ everywhere regardless of whether the user chooses to use different display units (mils, inches, or microns) in the GUI.  
One exception is that in `param` elements which pertain to geometry, the value includes units (e.g. `0.35mm` or `6mil`) - I don't know why, but that's how it is.

Reply:
But regardless, you were right about the scaling being wrong for the mm case here.